js/core.js

```diff
@@ -600,6 +636,39 @@ var ERMrest = (function (module) {
                 // add to referredBy of the key table
                 foreignKeyRef.key.table.referredBy._push(foreignKeyRef);
             }
+        },
+
+        // returns visible foreignkeys.
+        _visibleForeignKeys: function (context) {
+            var orders = -1;
+            try {
+                var annot = this.annotations.get(module._annotations.VISIBLE_FOREIGN_KEYS);
```

Comment:
You should use the `annotations.contains()` method to check whether this annotation exists, then do the `get()` call. Otherwise you will get `NotFoundException` here.